proc/put/put.go

```diff
@@ -34,28 +36,29 @@ type Proc struct {
 // language.
 type putRule struct {
 	typ         *zng.TypeRecord
-	clauseTypes []clauseType
-	// The clause numbers indexed by input column number that should be
-	// replaced, where -1 indicates no replacement.  Nil if there are no
-	// replacements.
-	replace []int
-	// The clause numbers that should be appeneded to the output record
-	// ordered left to right.  Nil if nothing need be appended.
-	append []int
-}
-
-type clauseType struct {
-	zng.Type
-	container bool
+	clauseTypes []zng.Type
+	step        step
 }
 
 func New(pctx *proc.Context, parent proc.Interface, clauses []expr.Assignment) (proc.Interface, error) {
+	for i, p := range clauses {
+		for j, c := range clauses {
+			if i != j && p.LHS.Equal(c.LHS) {
+				return nil, fmt.Errorf("put: multiple assignments to %s", p.LHS)
+			}
```

Comment:
Why do any checks when `i == j`?
```suggestion
			if i == j {
				continue
			}
			if p.LHS.Equal(c.LHS) {
				return nil, fmt.Errorf("put: multiple assignments to %s", p.LHS)
			}
```

Reply:
👍